core/src/main/scala/org/apache/spark/rdd/ReliableRDDCheckpointData.scala

```diff
@@ -31,12 +31,26 @@ import org.apache.spark.util.SerializableConfiguration
 private[spark] class ReliableRDDCheckpointData[T: ClassTag](@transient private val rdd: RDD[T])
   extends RDDCheckpointData[T](rdd) with Logging {
 
+  import CheckpointState._
+
   // The directory to which the associated RDD has been checkpointed to
   // This is assumed to be a non-local path that points to some reliable storage
-  private val cpDir: String =
-    ReliableRDDCheckpointData.checkpointPath(rdd.context, rdd.id)
+  private val cpDir: String = {
+    val _cpDir = ReliableRDDCheckpointData.checkpointPath(rdd.context, rdd.id)
       .map(_.toString)
-      .getOrElse { throw new SparkException("Checkpoint dir must be specified.") }
+      .getOrElse {
+      throw new SparkException("Checkpoint dir must be specified.")
+    }
+    val path = new Path(_cpDir)
+    val fs = new Path(_cpDir).getFileSystem(rdd.context.hadoopConfiguration)
+    if (!fs.mkdirs(path)) {
+      throw new SparkException("Failed to create checkpoint path " + path)
```

Comment:
That's how it was before. 

If checkpointing fails at the first time, `_cpDir` won't be deleted. Then the user may try to do it again, so we should allow checkpointing the same RDD even if `_cpDir` exists.